paasta_tools/flink_tools.py

```diff
@@ -361,3 +409,443 @@ def get_flink_overview_from_paasta_api_client(
         service=service,
         instance=instance,
     )
+
+
+def get_flink_instance_details(
+    metadata: Mapping[str, Any],
+    flink_config: Optional[Any],
+    flink_instance_config: Any,
+    service: str,
+    soa_dir: str = DEFAULT_SOA_DIR,
+) -> FlinkInstanceDetails:
+    """Collect Flink instance metadata and configuration details.
+
+    :param metadata: Kubernetes metadata from flink status
+    :param flink_config: Flink configuration (None if not running)
+    :param flink_instance_config: Flink instance config from yelpsoa-configs
+    :param service: Service name
+    :param soa_dir: SOA directory path
+    :returns: Dict with instance details
+    """
+    from paasta_tools.monitoring_tools import get_runbook
+    from paasta_tools.monitoring_tools import get_team
+
+    labels = metadata.get("labels", {})
+    annotations = metadata.get("annotations", {})
+
+    # Extract config SHA
+    config_sha = labels.get(paasta_prefixed("config_sha"))
+    if config_sha and config_sha.startswith("config"):
+        config_sha = config_sha[6:]
```

Comment:
non-blocking: i sorta like having the config sha prefixed with config since it makes it a lot clearer that it's not actually going to map to a git sha - but up to y'all since this is flink status code :p